From 4cb4b701e26adebeffca7075c8fc838973bbd8ce Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Wed, 3 Jun 2026 00:48:08 +0200
Subject: [PATCH] winegstreamer: Handle NULL filter in autoplug_query_cb.

---
 dlls/winegstreamer/wg_parser.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index 951f8fdde7c..df588d234ce 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -617,6 +617,11 @@ static gboolean autoplug_query_cb(GstElement *bin, GstPad *child,
     {
         result = gst_caps_new_empty();
         gst_query_parse_caps(query, &filter);
+        if (!filter)
+        {
+            gst_caps_unref(result);
+            return FALSE;
+        }
         for (i = 0; i < gst_caps_get_size(filter); i++)
         {
             if (!(features = gst_caps_get_features(filter, i))
-- 
2.54.0

