From 37c7bbebc4c2461408ac219afe0483ac64e73fd1 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Fri, 5 Jun 2026 13:46:23 +0300
Subject: [PATCH 1/2] win32u: Silence NtGdiDdDDIQueryStatistics spam

---
 dlls/win32u/d3dkmt.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/dlls/win32u/d3dkmt.c b/dlls/win32u/d3dkmt.c
index 38f7a53c3fb..465352d4943 100644
--- a/dlls/win32u/d3dkmt.c
+++ b/dlls/win32u/d3dkmt.c
@@ -764,7 +764,13 @@ NTSTATUS WINAPI NtGdiDdDDIQueryAdapterInfo( D3DKMT_QUERYADAPTERINFO *desc )
  */
 NTSTATUS WINAPI NtGdiDdDDIQueryStatistics( D3DKMT_QUERYSTATISTICS *stats )
 {
-    FIXME( "(%p): stub\n", stats );
+    static unsigned int once;
+
+    if (!once++)
+        FIXME( "(%p): stub\n", stats );
+    else
+        WARN( "(%p): stub\n", stats );
+
     return STATUS_SUCCESS;
 }
 
-- 
2.54.0

